test/test_autograd.py

```diff
@@ -7912,7 +7912,7 @@ def __new__(cls, data=None):
 
             @classmethod
             def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
-                if func == torch.ops.aten.alias:
+                if func.overload_packet == torch.ops.aten.alias:
```

Comment:
shouldn't this just be `func == torch.ops.aten.alias['']` (or whatever the syntax is for the alias overload)?

Reply:
sure so we could also change it to `if func == torch.ops.aten.alias.default:` (we use 'default' as the attribute to access '' overload name)

Reply:
Since it took a while for us to make this BC-breaking change, it seems to me that we should introduce some goop to smooth over the transition. The simplest thing we can do is overload the meaning of == on overloads so that they test equal to their overload packet (perhaps throwing a deprecation warning saying that this behavior will get removed in a later version).

We also need to publish guidance on when you should match against the overload_packet (this is still such a weird name to me haha) as opposed to the overload directly. Hot take: matching against the overload packet is probably the wrong thing to do in most cases.

Reply:
discussed offline that it's ok to just pull the plug